framework/src/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -89,7 +91,12 @@
         if (AnnotationUtils.areSameByClass(anno, DoubleVal.class)) {
             values = convertDoubleVal(anno, castType, castTo);
         } else if (AnnotationUtils.areSameByClass(anno, IntVal.class)) {
-            values = convertIntVal(anno, castType, castTo);
+            List<Long> longs = ValueAnnotatedTypeFactory.getIntValues(anno);
+            values = convertIntVal(longs, castType, castTo);
+        } else if (AnnotationUtils.areSameByClass(anno, IntRange.class)) {
+            Range range = ValueAnnotatedTypeFactory.getIntRange(anno);
+            List<Long> longs = getValuesFromRange(range, Long.class);
```

Comment:
Done. I've put the guard of `MAX_VALUES` inside the method so that we don't need guards everywhere.